Firestore/Source/Local/FSTLevelDBQueryCache.mm

```diff
@@ -239,23 +235,23 @@ - (nullable FSTQueryData *)queryDataForQuery:(FSTQuery *)query {
   // Note that this is a scan rather than a get because canonicalIDs are not required to be unique
   // per target.
   Slice canonicalID = StringView(query.canonicalID);
-  std::unique_ptr<Iterator> indexItererator(_db->NewIterator([FSTLevelDB standardReadOptions]));
+  std::unique_ptr<LevelDbTransaction::Iterator> indexItererator(
+      _db.current_transaction->NewIterator());
```

Comment:
Nearly all usages of `_db.current_transaction` just call `NewIterator`. Is it worth shortening this to cut down on the verbiage of this very common incantation?

Also, if `LevelDbTransaction.NewIterator` returned a `std::unique_ptr<LevelDbTransaction::Iterator>` this could be:

```c++
auto indexIterator = _db.NewIterator();
```
Or, in the short term:
```objc
auto indexIterator = [_db newIterator];
```

Reply:
In my ideal world, blocks given to `runTransaction` would accept a transaction as an argument, and then pass that around. I believe that runs into some type-casting ugliness with our current setup, so our compromise is to reach through to `_db` and grab the transaction. I don't think we should also add a second way of [implicitly] getting at the transaction by calling `newIterator` on `FSTLevelDB`. 

The other suggestion seems good though: switching `LevelDbTransaction::NewIterator` to return a `unique_ptr`. I'll make that change.

Reply:
This PR now includes the `unique_ptr` change.